src/openrct2-ui/windows/NewRide.cpp

```diff
@@ -45,6 +46,7 @@ static constexpr const int32_t WindowWidth = 601;
 static constexpr const int32_t RideListItemsMax = 384;
 static constexpr const int32_t RideTabCount = 6;
 static constexpr const int32_t GroupByTrackTypeWidth = 172;
+static std::string _filter;
```

Comment:
This should become a member variable of the window. You are clearing it in `OnOpen` so its not like it needs to be a local global variable (unless I'm not understanding somthing)

Reply:
I think you're right. I first attempted this when I had no knowledge about openrct2 code and was primarily mirroring the ObjectSelection window. I can also switch to u8string as well.